frontend/src/components/categoryList/categoryList.js

```diff
@@ -1,24 +1,102 @@
-import React from 'react';
+import React, {useRef} from 'react';
 import "./categoryList.css"
 import categoryService from '../../services/category.service';
 import {useEffect, useState} from "react";
+import {Link} from 'react-router-dom';
+import plus from '../../images/Plus.png'
+import minus from "../../images/Minus.png"
 
-function CategoryList() {
+function CategoryList({filter}) {
     const [categories, setCategories] = useState();
+    let show = useRef([]);
+    let mainContainer = "list"
+    let listContainer = "listItem"
+    let previousCategory = useRef(-1);
+    const [sign, setSign] = useState(-1);
+    let prodCat = useRef("");
+
+    function showSubcategories(event) {
```

Comment:
Can we pass event.target.src in showSubcategories method directly, so that we do not have to repeat this event.target.src through all this method ?

Reply:
It is not good to change src of image through this event.target.src...
Can we think about some other options ?